cupy/_core/core.pyx

```diff
@@ -2353,7 +2353,8 @@ cdef ndarray _array_from_nested_cupy_sequence(obj, dtype, shape, order):
 
 cdef ndarray _array_default(obj, dtype, order, Py_ssize_t ndmin):
     if order is not None and len(order) >= 1 and order[0] in 'KAka':
-        if isinstance(obj, numpy.ndarray) and obj.flags.f_contiguous:
+        if isinstance(obj, numpy.ndarray) and \
+           obj.flags.f_contiguous and not obj.flags.c_contiguous:
```

Comment:
This could be written more concisely using the [fnc](https://github.com/cupy/cupy/blob/7ca49c10f09b10d1330f5aaff042c98d02791b00/cupy/_core/flags.pyx#L11-L13) property. Maybe leaving it as-is avoids any overhead of calling the method, though?

```suggestion
        if isinstance(obj, numpy.ndarray) and obj.flags.fnc:
```